### PR TITLE
Big refactor of graph stuff

### PR DESCRIPTION
**Pull request by Sigfried**
- It should be more correct now
  - I think it's slower
  - The first two tests succeed now, but:
  - With lots of concepts (e.g., 20,000), it takes forever and the whole backend won't respond while it's processing
    - The problem isn't in connect_nodes anymore, it's in all_paths when it's looking for simple paths to root. Those paths can get long and there can be a lot of them. Maybe a strategy of looking for predecessors would work?
  - It needs a lot more inline documentation
- I may have fixed the duplicate test thing by putting the loop through tests inside the loop through environments rather than other way around.

### CODE REVIEW

**Comment by Sigfried** [comment]
I'm going to deploy this to dev

File: backend/db/ddl-20-concept_graph.jinja.sql [hunk 1]
```diff
@@ -1,14 +1,25 @@
 DROP TABLE IF EXISTS concept_graph CASCADE;
 
 CREATE TABLE IF NOT EXISTS concept_graph AS (
-     SELECT ancestor_concept_id AS source_id, descendant_concept_id AS target_id
+     SELECT ancestor_concept_id AS source_id,
```
Comment (joeflack4) [inline backend/db/ddl-20-concept_graph.jinja.sql:4]: Just confirming I've looked at `concept_graph` table for the first time.

File: backend/utils.py [hunk 2]
```diff
@@ -16,6 +17,12 @@
 from backend.config import CONFIG
 
 
+def powerset(iterable):
```
Comment (joeflack4) [inline backend/utils.py:20]: `powerset`: I don't see where this is used at the moment; fill me in if you want.
I like `chain`; don't use it that often. Haven't used `combinations` before.

File: frontend/tests/README.md [hunk 3]
```diff
@@ -11,13 +11,20 @@ what optimization is being performed and tested. Can take multiple values delimi
 |---------------------|------------------------------------------------------|:------------:|:--------:|
 | `as-is` or no value | Use code without attempting any special optimization |   yes  | yes |
 | `no-cache`          | Disables caching -- in DataCache.cachePut            |   yes  | yes |
-| `no-rxext-from-api` | RxNorm Extension codes left out of backend API results. See [below](#No-RxNorm-Extension-codes)|   no   | no  |
+| `no-rxext-from-api` | RxNorm Extension codes left out of backend API results. See [below](#no-rxnorm-extension-codes)|   no   | no  |
 
 
 ## Notes
 
 ### No RxNorm Extension codes
```
Comment (joeflack4) [inline frontend/tests/README.md:19]: Thanks for documenting.

**Pull request merged**